gio-subclass/src/auto/action.rs

```diff
@@ -0,0 +1,274 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir @ 1e22dcb+)
+// from gir-files (https://github.com/gtk-rs/gir-files @ 47c69e6)
+// DO NOT EDIT
+
+#[allow(unused_imports)]
+use libc::{c_int, c_char, c_uchar, c_float, c_uint, c_double,
+    c_short, c_ushort, c_long, c_ulong,
+    c_void, size_t, ssize_t, intptr_t, uintptr_t, time_t, FILE};
+
+#[allow(unused_imports)]
+use glib_ffi::{gboolean, gconstpointer, gpointer, GType};
+
+use free::*;
+
+#[cfg(any(feature = "v2_38", feature = "dox"))]
+use Error;
+use gio;
+use gio_ffi;
+use glib;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+use gobject_subclass::anyimpl::*;
+use gobject_subclass::object::*;
+
+
+pub trait ActionImpl: AnyImpl + 'static {
+
+    fn activate(&self, action: &gio::Action, parameter: Option<&glib::Variant>);
+
+    fn change_state(&self, action: &gio::Action, value: &glib::Variant);
+
+    fn get_enabled(&self, action: &gio::Action) -> bool;
+
+    fn get_name(&self, action: &gio::Action) -> Option<String>;
+
+    fn get_parameter_type(&self, action: &gio::Action) -> Option<glib::VariantType>;
+
+    fn get_state(&self, action: &gio::Action) -> Option<glib::Variant>;
+
+    fn get_state_hint(&self, action: &gio::Action) -> Option<glib::Variant>;
+
+    fn get_state_type(&self, action: &gio::Action) -> Option<glib::VariantType>;
+
+}
+
+any_impl!(ActionImpl);
+
+pub trait ActionImplStatic<T: ObjectType>: 'static {
+    fn get_impl<'a>(&self, imp: &'a T::ImplType) -> &'a ActionImpl;
+}
+
+struct ActionStatic<T: ObjectType>{
+    imp_static: *const ActionImplStatic<T>
+}
+
+// FIXME: Boilerplate
+
+unsafe extern "C" fn action_activate<T: ObjectType>
+(gptr: *mut gio_ffi::GAction, parameter: *mut glib_ffi::GVariant)
+{
+    floating_reference_guard!(gptr);
+    let klass = &**(gptr as *const *const ClassStruct<T>);
+    let interface_static = klass.get_interface_static(gio_ffi::g_action_get_type())
+                                     as *const ActionStatic<T>;
+    let instance = &*(gptr as *const T::InstanceStructType);
+    let imp = instance.get_impl();
+    let imp = (*(*interface_static).imp_static).get_impl(imp);
+    let wrap = from_glib_borrow(gptr);
+    imp.activate(&wrap, (if parameter.is_null() { None } else { Some(from_glib_none(parameter)) }).as_ref())
```

Comment:
You want `Some(&from_glib_borrow(parameter))` here

Reply:
Haha, I made the current implementation as per your previous review :).
I can do ```from_glib_borrow``` here. What's the difference?

Reply:
Potentially fewer copies :)